### PR TITLE
React Capstone - COVID-19 data tracking

### PR DESCRIPTION
**Pull request by Kayonga99**
## The following were implemented:

- No linter errors.
- Use correct git-flow.
- Select API.
- Home page
   - When the page loads, the webapp shows the list of categories that could be filtered out by some parameter, for example by the category name.
    - Along with the category name, you will display some numeric values per category that come from the API.
    - When the user clicks (or taps) on a category item, the application navigates to the details page.
- Details page

   - In the details page, the webapp retrieves data from the API to show detailed data in the category.
   - When the user clicks on the "Back" button (<), the user navigates to the home page.
- Create unit tests for pure functions in your Redux code.
- Create integration tests for your application using the React Testing Library.

### CODE REVIEW

File: src/components/CountryData.js [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+          </p>
+          <p>
+            People Recovered Today:
+            { todayRecovered }
+          </p>
+
+        </li>
+
+      </ul>
+    </div>
```
Comment (julie-ify) [inline src/components/CountryData.js:65]: - [x] We are trying to build a mobile app. That means that our app should maintain responsiveness across multiple mobile screen versions. Once I reduce the screen width to about less than 441px it starts to overflow. Kindly fix this.

- [x] Additionally, this lower part of this page continues with a white background once the properties are exhausted. That should be the case. The app background color should cover the entire screen for better UI.

  ando

File: src/components/Home.js [hunk 2]
```diff
@@ -0,0 +1,50 @@
+import React, { useEffect } from 'react';
+import { useSelector, useDispatch } from 'react-redux';
+import { Link } from 'react-router-dom';
+import { fetchData } from '../redux/covidData/covid';
+
+function Home() {
```
Comment (julie-ify) [inline src/components/Home.js:6]: - [x] Based on the requirements, the Home component should have a filtering mechanism, say a search box, with which users can filter countries on the homepage.

  gim

- [x] Based on the design being used, your Home page should have a banner section containing information about the countries listed on the home page

  fdd

File: README.md [hunk 3]
```diff
@@ -1,62 +1,58 @@
...
+npm start
+```
 
-**This is an example of how you may give instructions on setting up your project locally.**
-**Modify this file to match your project, remove sections that don't apply. For example: delete the testing section if the currect project doesn't require testing.**
+```
+http://localhost:3000
+```
 
+- [click here! to see live demo]()
```
Comment (julie-ify) [inline README.md]: - [x] Add a live demo link

- [ ] Record a video presentation of the project and include the link in your readme documentation.

File: src/__test__/fetchCovid.test.js [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+    await fetchData().then((data) => {
+      expect(data[0].deaths).toBe(3049);
+    });
+  });
+  it('test fetch covid cases', async () => {
+    await fetchData().then((data) => {
+      expect(data[0].cases).toBe(157015);
+    });
+  });
+});
```
Comment (julie-ify) [inline src/__test__/fetchCovid.test.js:26]: - [x] Good job writing tests for the data fetch, however you still need to write more integration tests covering multiple components using the React Testing Library.

- [x] Kindly include more unit tests for several components.

File: README.md [hunk 5]
```diff
@@ -68,8 +64,4 @@ Give a ⭐️ if you like this project!
 - Inspiration
 - etc
```
Comment (julie-ify) [inline README.md]: - [ ] Your acknowledgments section in the documentation should contain appropriate credit to Nelson Sakwa whose design was used.

**Review by julie-ify — CHANGES_REQUESTED** [review]
## Hi @Kayonga99,

Good job so far!

## Highlights!
- Good feature branch name ✔️ 
- No linter errors ✔️ 
- Descriptive pull request ✔️ 
- Descriptive commit messages ✔️ 
- Good code organization and working app ✔️
- Descriptive README ✔️ 

Hello ✋, you have done well reaching this point of the milestone 🤝 . But there are some issues that you still need to work on to go to the next project but you are almost there!

# Status: Required Changes ♻️⛔️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread and tag me `@julie-ify` if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: src/__test__/fetchCovid.test.js [hunk 6]
```diff
@@ -0,0 +1,26 @@
...
+  it('test fetch deaths', async () => {
+    await fetchData().then((data) => {
+      expect(data[0].deaths).toBe(3049);
+    });
+  });
+  it('test fetch covid cases', async () => {
+    await fetchData().then((data) => {
+      expect(data[0].cases).toBe(157015);
+    });
+  });
```
Comment (1GORDON) [inline src/__test__/fetchCovid.test.js:25]: Great job @Kayonga99 🤝

- [ ] Just like the previous reviewer pointed out, kindly add more integration tests covering multiple components using the React Testing Library.
- [ ] Kindly include more unit tests for several components.

File: README.md [hunk 7]
```diff
@@ -1,62 +1,58 @@
...
-- LinkedIn: [LinkedIn](https://linkedin.com/in/linkedinhandle)
-
-## 🤝 Contributing
-
-Contributions, issues, and feature requests are welcome!
-
-Feel free to check the [issues page](../../issues/).
+if you are interested in learning more about webpack click the link below
 
 ## Show your support
```
Comment (1GORDON) [inline README.md:57]: Great job @Kayonga99 🤝

- [ ] As the previous reviewer stated, kindly add the acknowledgment section in your readme file and give credit to Nelson Sakwa whose design was used.

File: src/App.js [hunk 8]
```diff
@@ -0,0 +1,19 @@
...
+  return (
+    <Router>
+      <Navbar />
+      <Routes>
+        <Route path="/" element={<Home />} />
+        <Route path="/country/:name" element={<CountryData />} />
+      </Routes>
+    </Router>
+  );
+}
```
Comment (1GORDON) [inline src/App.js:17]: Great job @Kayonga99 🤝

- [ ] [Optional] Kindly use ES6 syntax for all your functions and methods across your project, as this is regarded as a good practice for Javascript.

**Review by 1GORDON — CHANGES_REQUESTED** [review]
##
## Hi @Kayonga99 ✋

Good job so far!

## Highlights!
- Followed Gitflow ✔️ 
- Good commit messages ✔️
- Linters are passing ✔️
- Descriptive Pull request ✔️
- Professional readme ✔️

Hello ✋, you have done well reaching this point of the milestone 🤝 . But there are some issues that you still need to work on to go to the next project but you are almost there!

# Status: Required Changes ♻️⛔️

- [ ] Kindly record a video presentation of the project and include the link in your readme documentation. You may check your project documentation section in your [project requirements](https://github.com/microverseinc/curriculum-react-redux/blob/main/capstone/react_capstone.md).

_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread and tag me `@1GORDON` if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by ShahierNashaat — APPROVED** [review]
Hi @Kayonga99, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## To Highlight 💯 

- All tests are passing ✔️
- Correct use of GitFlow ✔️
- Descriptive commit messages ✔️
- Professional looking pull request ✔️
- Your readme looks professional ✔️
- Good overall design ✔️
- Your code is clean and well documented ✔️

# Approved :ok_hand: :ok_hand: :ok_hand:

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**